### PR TITLE
Let's fix some bugs !

### PR DESCRIPTION
**Pull request by techtastet**
- I added some tests to check if everything is okay before merge
- I fixed the crash when an update in MySQL is executed (updatedAt field was the reason)
- And I fixed an oversight in the types file :)

Hope all this will be useful in the future ;)

### CODE REVIEW

File: tests/connection.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Database } from "../mod.ts";
+import { config } from "https://deno.land/x/dotenv/mod.ts";
+
+const ConnectionMySQL = (): Database => {
```
Comment (eveningkid) [inline tests/connection.ts:4]: - The naming could be confusing as it implies this is a class, although when used you don't have to instantiate it. `getMySQLConnection()` would be more descriptive.
- Would be great if the options could be overwritten —if we ever need to do so:
```js
const getMySQLConnection = (options? = {}): Database => {
  // ...later merges options with { database: ... }
};
```

File: tests/units/MySQL/connection.test.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { ConnectionMySQL } from "../../connection.ts";
+import { assertEquals } from "../../deps.ts";
+
+Deno.test("MySQL connection", async function () {
+  const connection = ConnectionMySQL();
```
Comment (eveningkid) [inline tests/units/MySQL/connection.test.ts:5]: Every time in use, the name/import should be replaced then! 😄

File: tests/units/MySQL/models.test.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import { DataTypes, Model } from "../../../mod.ts";
+import { ConnectionMySQL } from "../../connection.ts";
+import { assertEquals } from "../../deps.ts";
+
+Deno.test("MySQL One Model", async function () {
+  const connection = ConnectionMySQL();
```
Comment (eveningkid) [inline tests/units/MySQL/models.test.ts:6]: Here too.

File: tests/units/MySQL/models.test.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    departure: "Paris",
+    destination: "Tokyo",
+  });
+
+  // Get
+  const result = await Flight.where({ departure: "Paris" }).first();
+
+  await connection.close();
+
+  delete result.updatedAt;
```
Comment (eveningkid) [inline tests/units/MySQL/models.test.ts:38]: Would the following still be necessary if we set `Flight.timestamps = false`?

I wonder actually. From what I remember, these fields are only added when `timestamps = true`.

File: tests/units/Relationships/Foreinkey.test.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
+import { DataTypes, Model, Relationships } from "../../../mod.ts";
+import { ConnectionMySQL } from "../../connection.ts";
+import { assertEquals } from "../../deps.ts";
+
+class Owner extends Model {
+  static table = "foreinkeyowners";
```
Comment (eveningkid) [inline tests/units/Relationships/Foreinkey.test.ts:6]: `foreiGnkeyowners`, there are missing `g`s in that file (in the filename as well).

Just need to replace these

File: tests/units/Relationships/Foreinkey.test.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    id: {
+      type: DataTypes.INTEGER,
+      primaryKey: true,
+    },
+    name: DataTypes.STRING,
+  };
+}
+
+class Business extends Model {
+  static table = "foreinkeybusinesses";
```
Comment (eveningkid) [inline tests/units/Relationships/Foreinkey.test.ts:18]: Same here (I just pin it so you wouldn't have to resend another commit later haha)

File: tests/units/Relationships/Foreinkey.test.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    name: DataTypes.STRING,
+    ownerId: Relationships.belongsTo(Owner),
+  };
+
+  static owner() {
+    return this.hasOne(Owner);
+  }
+}
+
+Deno.test("MySQL - Forein Key test", async function () {
```
Comment (eveningkid) [inline tests/units/Relationships/Foreinkey.test.ts:34]: Same

File: tests/units/Relationships/Foreinkey.test.ts [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+  const connection = ConnectionMySQL();
```
Comment (eveningkid) [inline tests/units/Relationships/Foreinkey.test.ts:35]: Remember to change the name here as well

File: tests/units/Relationships/Foreinkey.test.ts [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+  await Business.create({
+    id: "1",
+    name: "Parisian Café",
+    ownerId: "1",
+  });
+
+  const OwnerTest = await Business.where("id", "1").owner();
+
+  delete OwnerTest.updatedAt;
```
Comment (eveningkid) [inline tests/units/Relationships/Foreinkey.test.ts:54]: Same question as in the other file. Is it needed if `timestamps = false`?

File: tests/units/queries/update.test.ts [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+      primaryKey: true,
+      autoIncrement: true,
+    },
+    title: DataTypes.STRING,
+    content: DataTypes.TEXT,
+  };
+}
+
+Deno.test("MySQL - Update test", async function () {
+  const connection = ConnectionMySQL();
```
Comment (eveningkid) [inline tests/units/queries/update.test.ts:20]: Same

File: tests/units/queries/update.test.ts [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+  await Article.create({
+    title: "Hello world !",
+    content: "first articlE !",
+  });
+
+  await Article.where({ id: 1 }).update({ content: "first article !" });
+
+  const article = await Article.where({ id: 1 }).first();
+
+  delete article.updatedAt;
```
Comment (eveningkid) [inline tests/units/queries/update.test.ts:35]: ...same! 😄

File: tests/units/Relationships/Foreinkey.test.ts [hunk 12]
```diff
@@ -0,0 +1,66 @@
+import { DataTypes, Model, Relationships } from "../../../mod.ts";
```
Comment (eveningkid) [inline tests/units/Relationships/Foreinkey.test.ts:1]: Just wondering what's the filename coming from. There are capital letters in this one but not in the query test suite.

I think it is common to keep these lowercased as you did with the queries :)

**Review by eveningkid — CHANGES_REQUESTED** [review]
Hi Thomas,

Thanks for all your efforts :)

Almost there, I left a few comments.
Shouldn't take too long to make these changes (it's just to make it all blend perfectly).

We can merge after these changes are done.

Keep in touch!

**Comment by techtastet** [comment]
done @eveningkid :)

**Review by eveningkid — APPROVED** [review]
Hi Thomas,

There are still a few typos so I'll make a commit after merging.

Thank you for your work, I will merge now :)

**Pull request merged**